src/async-components/views/dialogs/eventindex/DisableEventIndexDialog.js

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import * as sdk from '../../../../index';
+import PropTypes from 'prop-types';
+import dis from "../../../../dispatcher";
+import { _t } from '../../../../languageHandler';
+
+import SettingsStore, {SettingLevel} from "../../../../settings/SettingsStore";
+import EventIndexPeg from "../../../../indexing/EventIndexPeg";
+
+/*
+ * Allows the user to disable the Event Index.
+ */
+export default class DisableEventIndexDialog extends React.Component {
+    static propTypes = {
+        onFinished: PropTypes.func.isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            disabling: false,
+        };
+    }
+
+    _onDisable = async () => {
+        this.setState({
+            disabling: true,
+        });
+
+        await SettingsStore.setValue('enableEventIndexing', null, SettingLevel.DEVICE, false);
+        await EventIndexPeg.deleteEventIndex();
+        this.props.onFinished();
+        dis.dispatch({ action: 'view_user_settings' });
+    }
+
+    render() {
+        const BaseDialog = sdk.getComponent('views.dialogs.BaseDialog');
+        const Spinner = sdk.getComponent('elements.Spinner');
+        const DialogButtons = sdk.getComponent('views.elements.DialogButtons');
+
+        return (
+            <BaseDialog className='mx_ManageEventIndexDialog'
+                onFinished={this.props.onFinished}
+                title={_t("Are you sure?")}
+            >
+                {_t("If disabled, messages from encrypted rooms won't appear in search results.")}
+                {this.state.disabling ? <Spinner /> : <div />}
+                <DialogButtons
```

Comment:
For this one, where we want something like:

<img width="200" alt="2020-01-24 at 13 34" src="https://user-images.githubusercontent.com/279572/73073011-4de05300-3eae-11ea-881e-b540a557f0a1.png">

I don't think we have a standard easy way to style the "cancel" button with red text and no background. Here's [an approach](https://github.com/matrix-org/matrix-react-sdk/blob/7de5f09f9f1844fa9fc0d1a84dcf7671aac3cfba/src/components/structures/auth/CompleteSecurity.js#L143-L157) I used recently to achieve it. Since we don't have `DialogButtons` with that approach, you'll a little CSS to space out the buttons, such as [this block](https://github.com/matrix-org/matrix-react-sdk/blob/7de5f09f9f1844fa9fc0d1a84dcf7671aac3cfba/res/css/structures/auth/_CompleteSecurity.scss#L40-L51).

(We should generalised this stuff it seems, but maybe we'll wait for it to appear a few more times first.)

Reply:
Can't we just add a `warning` class and appropriate style for that class? This way I can use the prop that I added to the `DialogButtons`

Reply:
Yes, feel free to use a `warning` class for this style. Adding some more CSS [near the dialog button styles](https://github.com/matrix-org/matrix-react-sdk/blob/develop/res/css/_common.scss#L425-L429) is probably what would be needed.